src/rgw/rgw_rest_user.cc

```diff
@@ -646,7 +647,11 @@ void RGWOp_Key_Create::execute(optional_yield y)
   RESTArgs::get_string(s, "secret-key", secret_key, &secret_key);
   RESTArgs::get_string(s, "key-type", key_type_str, &key_type_str);
   RESTArgs::get_bool(s, "generate-key", true, &gen_key);
+  RESTArgs::get_string(s, "tenant", tenant_name, &tenant_name);
 
+  if (!tenant_name.empty()) {
+    uid.tenant = tenant_name;
+  }
```

Comment:
**Suggestion:** instead of setting the value explicitly we can also pass it with the constructor
```c++
    RESTArgs::get_string(s, "tenant", tenant_name, &tenant_name);
    rgw_user uid(tenant_name, uid_str);
```

Reply:
yes I agree, but other places they follow this format

Reply:
Okay, I think can change it. but this works as well. 👍